### PR TITLE
Cover SKR in fast-integration tests

### PR DESCRIPTION
**Pull request by crabtree**
**Description**

Changes proposed in this pull request:

- Cover SKR in fast-integration tests

**Related issue(s)**

### CODE REVIEW

**Review by klaudiagrz — APPROVED** [review]

File: tests/fast-integration/utils/index.js [hunk 1]
```diff
@@ -692,7 +715,86 @@ function getEnvOrThrow(key) {
...
+      
+      function done(fn, arg) {
+          clearTimeout(th);
+          clearInterval(ih);
+          fn(arg);
+      }
+  });
+}
+
+async function ensureApplicationMapping(name, ns) {
```
Comment (pbochynski) [inline tests/fast-integration/utils/index.js:739]: I would rather keep it in commerce-mock fixture. Utils are a more generic functions. The same with patching application gateway.

File: tests/fast-integration/kyma-environment-broker/client.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+  constructor(config) {
+    this.host = config.host;
+    this.clientID = config.clientID;
+    this.clientSecret = config.clientSecret;
+    this.globalAccountID = config.globalAccountID;
+    this.subaccountID = config.subaccountID;
+    this.planID = config.planID
+    this.serviceID = KYMA_SERVICE_ID;
+    
+    this._token = void 0;
```
Comment (pbochynski) [inline tests/fast-integration/kyma-environment-broker/client.js:42]: I don't like `void 0`. What is the purpose of this explicit assignment? For me just an unnecessary line that makes it harder to understand (I had to read 3 articles about why people used this construction in the previous version of ECMA script).

File: tests/fast-integration/kyma-environment-broker/helpers.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  const operationID = resp.operation
+  const shootName = resp.dashboard_url.split(".")[1];
+  debug(`Operation ID ${operationID}`, `Shoot name ${shootName}`);
+
+  await ensureOperationSucceeded(keb, instanceID, operationID);
+
+  const shoot = await gardener.getShoot(shootName);
+  debug(`Compass ID ${shoot.compassID}`);
+
+  fs.writeFileSync(SHOOT_KUBECONFIG_PATH, shoot.kubeconfig);
```
Comment (pbochynski) [inline tests/fast-integration/kyma-environment-broker/helpers.js:26]: I would consider not storing shoot kubeconfig in the file. Wouldn't it be cleaner if you just initialize k8s client from the string?

**Review by pbochynski — CHANGES_REQUESTED** [review]

File: tests/fast-integration/utils/index.js (hunk 1 above)
Comment (crabtree) [reply tests/fast-integration/utils/index.js:739]: Requires further refactoring of utils and I think we should extract from utils all k8s related functions to separate client initialised from kubeconfig.

File: tests/fast-integration/kyma-environment-broker/client.js (hunk 2 above)
Comment (crabtree) [reply tests/fast-integration/kyma-environment-broker/client.js]: Replaced with undefined. I like to have all properties declared in the class constructor just for transparency.

File: tests/fast-integration/kyma-environment-broker/helpers.js (hunk 3 above)
Comment (crabtree) [reply tests/fast-integration/kyma-environment-broker/helpers.js]: done

**Review by pbochynski — APPROVED** [review]

**Review by klaudiagrz — APPROVED** [review]

**Pull request merged**